src/Server.ts

```diff
@@ -0,0 +1,85 @@
+import assert from 'assert'
+import express, { Express } from 'express'
+import * as http from 'http'
+
+import Logger from 'helpers/Logger'
+import { Request, ParamsDictionary, Response, NextFunction } from 'express-serve-static-core'
+
+const log = new Logger('server')
+export interface Params {
+  port: number
+}
+
+export class Server {
+  private _port: number
+  private _server: http.Server | null = null
+
+  constructor (params: Params) {
+    const { port } = params
+    this._port = port
+  }
+
+  public start (): Promise<void> {
+    log.debug(`Starting server on port ${this._port}...`)
+    assert(this._server === null, 'Server was already started')
+    const app = express()
+
+    // Register endpoints
+    this._registerEndpoint(app)
+
+    // Register middleware
+    this._registerMiddleware(app)
+
+    // Create server
+    this._server = http.createServer(app)
```

Comment:
Any reason you went with `http.createServer` over the convenient `app.listen`?

Reply:
Not really, I added it when trying to log errors, I think that as for now is equivalent

Reply:
If you want I can change it, but I kind of like to create the server, subscribe and then listen, instead of creating the server by listening and then subscribe.

I wonder if the errors are caught in the `on` also if I use app.listen

Reply:
app.listen returns that same server, so need to listen afterwards
Doesn't really matter, was just wondering why
Leave it